### PR TITLE
feat: Added `rfft` to jax numpy frontend

### LINKED ISSUE #22878: rfft

### PR DESCRIPTION
**Pull request by AbdullahSabry**
Closes #22878

### CODE REVIEW

**Comment by AbdullahSabry** [comment]
Hey @abdulasiraj 

Had to deal with some issues in the tensorflow backend with unsupported types going into `fft`. Got that resolved with some datatype handling. Please review the latest changes when possible, Thanks!

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by AbdullahSabry** [comment]
@ivy-leaves I have mentioned the issue in the PR description but it is #22878

File: ivy/functional/frontends/jax/numpy/fft.py [hunk 1]
```diff
@@ -48,3 +48,21 @@ def ifft2(a, s=None, axes=(-2, -1), norm=None):
...
+
+
+@with_unsupported_dtypes({"2.5.2 and below": ("float16", "bfloat16")}, "paddle")
+@to_ivy_arrays_and_back
+def rfft(a, n=None, axis=-1, norm=None):
+    if norm is None:
+        norm = "backward"
+    if n is None:
+        n = len(a)
+    if ivy.current_backend_str() == "tensorflow":
```
Comment (abdulasiraj) [inline ivy/functional/frontends/jax/numpy/fft.py:60]: Hi @AbdullahSabry,
it's not the ideal way to deal with tensorflow backend. I think either we've to use `with_unsupported_dtypes` for tensorflow or restricted the data types in the tensorflow backend for fft function.

**Review by abdulasiraj — CHANGES_REQUESTED** [review]
Everything else look great. Thanks :)

File: ivy/functional/frontends/jax/numpy/fft.py (hunk 1 above)
Comment (AbdullahSabry) [reply ivy/functional/frontends/jax/numpy/fft.py]: Pushed changes to make the casting happen on all backends to avoid confusion

**Comment by AbdullahSabry** [comment]
Someone already implemented this while the PR was active. Closing this

**Pull request closed without merging**